sherpa/astro/xspec/__init__.py

```diff
@@ -196,7 +201,7 @@ def __init__(self, filename, name='xstbl', parnames=(),
 
         # make translation table to turn reserved characters into '_'
         bad = string.punctuation+string.whitespace
```

Comment:
`punctuation` and `whitespace` are still attributes of `sting` in Python3, as far as I can tell. The problem with `(xs)tablemodel` is that the code tries to instantiate the xspec table model class, and falls back to the regular table model if it doesn't work, silently. The Python 3 migration is a lot of work already, and I am trying to be strict and resist the urge to fix the issues I stumble upon.